cdap-api/src/main/java/co/cask/cdap/api/service/http/HttpContentProducer.java

```diff
@@ -61,6 +63,7 @@ public long getContentLength() {
    *
    * @throws Exception if there is any error
    */
+  @TransactionPolicy(TransactionControl.IMPLICIT)
```

Comment:
Similarly, you might want to annotate the `nextChunk` method with `EXPLICIT` policy. 

For the `getContentLength` method, I think currently it is always called without transaction, since it is called from the netty-http `HttpResponder.sendContent` method, which in turn is called from the `DelayedHttpServiceResponder.execute` method, which is always non-transactional.